### PR TITLE
Issue:  Secure Trendy Website

### PR DESCRIPTION
**Pull request by mufudzimasaire**
### Overview

Related GitHub issue: https://github.com/shiftcommerce/trendy-golf/issues/161

PR moves the Content Security Policy from the reference app into `shift-next-routes`. It also refactors decouples the `FeaturePolicy` from the `CSP` as well as adds HTTP security headers.

### QA

1.) Checkout branch `issue/161_secure_trendy` and boot app.

2.) Using Ngrok as a proxy, test the site passes:

proxy

- https://securityheaders.com

security headers results

- https://www.immuniweb.com

immuniweb results

immuniweb warnings

### Code Review

All updated files

### Dependencies Review

None

## ENV variables to be added

N/A

### CODE REVIEW

File: packages/shift-next-routes/src/middleware/content-security-policy.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const formattedImageHosts = options.imageHosts ? options.imageHosts.split(',') : []
+  const formattedScriptHosts = options.scriptHosts ? options.scriptHosts.split(',') : []
+  
+  server.use(helmet.contentSecurityPolicy({
+    directives: {
+      defaultSrc: ["'self'"],
+      // Allow inline style attributes
+      styleSrc: ["'self'", "'unsafe-inline'"],
+      // Only allow first party images or from configured external hosts
+      // TODO: data: is insecure, currently this is a workaround for having inline SVGs
```
Comment (mufudzimasaire) [inline packages/shift-next-routes/src/middleware/content-security-policy.js]: seems this is still required as we still have some inline SVGs

File: packages/shift-next-routes/src/middleware/content-security-policy.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      styleSrc: ["'self'", "'unsafe-inline'"],
+      // Only allow first party images or from configured external hosts
+      // TODO: data: is insecure, currently this is a workaround for having inline SVGs 
+      // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+      imgSrc: ["'self'", 'data:'].concat(formattedImageHosts),
+      // Only allow first party scripts and those from Stripe. Unsafe inline is
+      // required as that's how we load Stripe
+      // TODO: Added 'unsafe-eval' in order for stripe to load correctly and to stop an webpack error "Uncaught TypeError: __webpack_require__(...) is not a function"
+      // for react index.js next and next-dev.  Is this approach safe?
+      scriptSrc: ["'self'", "'unsafe-inline'", 'https://js.stripe.com', 'https://*.paypal.com', 'https://*.paypalobjects.com', "'unsafe-eval'"],
```
Comment (mufudzimasaire) [inline packages/shift-next-routes/src/middleware/content-security-policy.js]: Screenshot 2019-05-30 at 14 29 50

**Comment by mufudzimasaire** [comment]
## QA on Staging

Screenshot 2019-06-06 at 11 19 10

File: packages/reference/next.config.js [hunk 3]
```diff
@@ -3,7 +3,9 @@ const withSass = require('@zeit/next-sass')
 const withCSS = require('@zeit/next-css')
 const nextRuntimeDotenv = require('next-runtime-dotenv')
 const withTM = require('next-transpile-modules')
-const withBundleAnalyzer = require('@zeit/next-bundle-analyzer')
+const withBundleAnalyzer = require('@next/bundle-analyzer')({
+  enabled: process.env.ANALYZE === 'true'
```
Comment (marcgear) [inline packages/reference/next.config.js]: could we get a tighter name for this env var 'ANALYZE_BUNDLE' or something perhaps?

Comment (marcgear) [reply packages/reference/next.config.js]: oh I see elsewhere there's a BUNDLE_ANALYZE - perhaps we have more than one env var here

**Comment by mufudzimasaire** [comment]
### QA before merge

- Deployed branch to heroku staging

security headers check

*******

ImmuniWeb check

Missing Privacy Policy

**Pull request merged**